plugins/modules/podman_play.py

```diff
@@ -249,12 +249,13 @@ def discover_pods(self):
         if self.module.params['kube_file']:
             if HAS_YAML:
                 with open(self.module.params['kube_file']) as f:
-                    pod = yaml.safe_load(f)
-                if 'metadata' in pod:
-                    pod_name = pod['metadata'].get('name')
-                else:
-                    self.module.fail_json(
-                        "No metadata in Kube file!\n%s" % pod)
+                    pods = list(yaml.safe_load_all(f))
+                for pod in pods:
+                    if 'metadata' in pod:
+                        pod_name = pod['metadata'].get('name')
```

Comment:
Actually it may be either `Pod` or `Deployment` if I understand correctly

Reply:
Update discover_pods() and tests. Please have a look!

Reply:
BTW, I think we can use [`podman kube down`](https://docs.podman.io/en/stable/markdown/podman-kube-down.1.html) to delete containers and pods defined in kube files. It might make it simple for this part (discover_pods and remove_associated_pods).

Reply:
Yeah, totally. This is much better way to delete all.

Reply:
I'll try to fix this point by using `podman kube play --down` in https://github.com/containers/ansible-podman-collections/pull/727 then.